.github/workflows/deploy.yaml

```diff
@@ -6,48 +6,30 @@ on:
       - main
   workflow_dispatch:
 
-jobs:
-  build:
-    name: Build Docusaurus
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
-        with:
-          fetch-depth: 0
-      - uses: actions/setup-node@v4
-        with:
-          node-version: 20
-          cache: npm
+env:
+  NODE_OPTIONS: --max_old_space_size=20480
+  GITHUB_REPOSITORY: ${{ github.repository }}
 
-      - name: Install dependencies
-        run: npm install --frozen-lockfile
-        
-      - name: Build specs and website
-        run: |
-          npm run build
-          npm run build:docs
-
-      - name: Upload Build Artifact
-        uses: actions/upload-pages-artifact@v3
-        with:
-          path: build
-
-  deploy:
-    name: Deploy to GitHub Pages
-    needs: build
-
-    # Grant GITHUB_TOKEN the permissions required to make a Pages deployment
+jobs:
+  deploy-gh-pages:
     permissions:
-      pages: write # to deploy to Pages
-      id-token: write # to verify the deployment originates from an appropriate source
-
-    # Deploy to the github-pages environment
-    environment:
-      name: github-pages
-      url: ${{ steps.deployment.outputs.page_url }}
+      pages: write
+      id-token: write
+      contents: write
 
     runs-on: ubuntu-latest
     steps:
+      - uses: actions/checkout@v4
+      - name: Use Node.js 22
+        uses: actions/setup-node@v4
+        with:
+          node-version: '22'
+      - run: npm ci
+      - run: npm run build
+      - run: npm run build:docs
       - name: Deploy to GitHub Pages
-        id: deployment
-        uses: actions/deploy-pages@v4
\ No newline at end of file
+        uses: peaceiris/actions-gh-pages@v4
```

Comment:
would be cool if we could stay on actions I think for long term usability

Reply:
I'm not sure what you mean here.  Can you clarify?